### ISSUE TITLE
Transition doesn't fire when changing state in an animation callback without manual change detection.

### ISSUE THREAD

**Issue by james-world** [angular/angular#13034]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
When changing the property that an animation trigger is bound to in an animation 'done' callback, an expected subsequent animation trigger bound to the same property does not fire in response to the new property value.

A workaround is to manually trigger change detection.

**Expected behavior**
The subsequent animation trigger should fire when state is changed in an animation callback without the need to invoke change detection.

**Minimal reproduction of the problem with instructions**
Full minimal repro is available at https://github.com/james-world/anim-bug

Run the code, and you will see the text "Text 1: I will fade out" fade out, and then the text "Text 2: I will fade in" appear.

But if you comment out line 70 of https://github.com/james-world/anim-bug/blob/master/src/app/app.component.ts (which fires manual change detection) then the Text 2 line does not fade in.

So the bug is exhibited by commenting out line 70 of app.component.ts.

**What is the motivation / use case for changing the behavior?**
Possibly this is a bug; if not it is a non-trivial step to fire the change-detection that isn't documented.

**Please tell us about your environment:**
Windows 10, Chrome 54, VS Code, NPM.

* **Angular version:** 2.1.0

* **Browser:** [ Chrome 54 | Firefox | IE 11 | Edge  ] (Suspect all affected)
 
* **Language:**  [ TypeScript 2.06 ]

* **Node (for AoT issues):** `node --version` =   6.9.1

**Comment by ericmartinezr** [comment]
This was fixed by https://github.com/angular/angular/pull/12774/commits/971407c5d00c3c030d43089aceff32952d5cbbbf, so you may try upgrading to latest. According to the [changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md#220-upgrade-firebooster-2016-11-14) that fix landed with 2.2.0

**Comment by james-world** [comment]
Yes, I can confirm that this is fixed in 2.2.0. Thanks.
